streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorStateManager.java

```diff
@@ -358,6 +276,9 @@ public void close(Map<TopicPartition, Long> ackedOffsets) throws IOException {
     // write the checkpoint
     @Override
     public void checkpoint(final Map<TopicPartition, Long> ackedOffsets) {
+        for (final StateRestorer activeStateRestorer : activeStateRestorers) {
+            checkpointedOffsets.put(activeStateRestorer.partition(), activeStateRestorer.restoredOffset());
```

Comment:
Could we just call this in line 291 below instead of maintaining a list of `activeStateRestorers` ?